### PR TITLE
feat(Artwork): shallow partner and artist in Artwork data when possible

### PR DESCRIPTION
**Pull request by araujobarret**
### Description

Co-authored with @joeyAghion so this PR has Joey's stamp of certified quality 🎆 

Doing some datadog investigation we noticed when we show Artworks in Rails we were hitting /partner and /artist for data that already comes packed with the Artwork. I tested the app, it looks solid but in the QA tomorrow might be cool to test it more extensively.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- feat: shallow some artwork props from artist and partner that come from the Artwork data

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (feat: shallow some artwork props from artist and partner that come from the Artwork data - araujobarret)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 696e0148c19ea8cb3a390b0cc5eddcf901514eee
</p>

**Review by gkartalis — COMMENTED** [review]
Question: what we should look for while testing?

**Review by gkartalis — APPROVED** [review]
Question: what we should look for while testing?

File: src/app/Scenes/CollectionsByCategory/CollectionRail.tsx [hunk 1]
```diff
@@ -89,9 +89,6 @@ const fragment = graphql`
     slug @required(action: NONE)
 
     artworksConnection(first: 10, sort: "-decayed_merch") {
-      counts @required(action: NONE) {
-        total
-      }
```
Comment (joeyAghion) [inline src/app/Scenes/CollectionsByCategory/CollectionRail.tsx:94]: 👍

**Review by joeyAghion — APPROVED** [review]
Since we worked on this, I also added partners' `inquireable` property to that model's short JSON (https://github.com/artsy/gravity/pull/18353), meaning it should be available within artworks' `partner(shallow: true)` properties. That could be a good follow-up, or added here if easy.

**Pull request merged**